src/zenml/cli/formatter.py

```diff
@@ -0,0 +1,147 @@
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+
+from typing import Dict, Iterable, Iterator, Optional, Sequence, Tuple
+
+from click import formatting
+from click._compat import term_len
+
+
+def measure_table(rows: Iterable[Tuple[str, ...]]) -> Tuple[int, ...]:
```

Comment:
I'm going to write this here, but I think all functions in this file could benefit from some docstrings and (especially the `write_dl` method some inline comments that explain what's happening as I couldn't easily follow it